app/assets/javascripts/pageflow/editor/models/text_track_file.js

```diff
@@ -0,0 +1,23 @@
+pageflow.TextTrackFile = pageflow.HostedFile.extend({
+  defaults: {
+    configuration: {
+      kind: 'captions'
+    }
+  },
+
+  initialize: function(attributes, options) {
+    pageflow.UploadedFile.prototype.initialize.apply(this, arguments);
+    if(this.isNew()) {
+      this.configuration.set('srclang', this.extractLanguageCodeFromFilename());
+      if(this.configuration.get('srclang' !== 'null')){
+        this.configuration.set('label',
+                               I18n.t('pageflow.languages.' + this.configuration.get('srclang')));
```

Comment:
Storing translations in the database, breaks localization for multi-lang teams and prevents switching entry locale. We can use a `placeholder` in the form view and use a I18n generated label as a default value.